packages/api/lambdas/write-db-dlq-records-to-s3.js

```diff
@@ -1,14 +1,90 @@
+//@ts-check
+
 'use strict';
 
 const get = require('lodash/get');
 const uuidv4 = require('uuid/v4');
 
 const log = require('@cumulus/common/log');
+const { isEventBridgeEvent } = require('@cumulus/common/lambda');
 const { s3PutObject } = require('@cumulus/aws-client/S3');
 const { parseSQSMessageBody } = require('@cumulus/aws-client/SQS');
 const { getMessageExecutionName } = require('@cumulus/message/Executions');
 const { unwrapDeadLetterCumulusMessage } = require('@cumulus/message/DeadLetterMessage');
+const { getCumulusMessageFromExecutionEvent } = require('@cumulus/message/StepFunctions');
+
+/**
+ *
+ * @typedef {import('@cumulus/types').MessageGranule} MessageGranule
+ * @typedef {{granules: Array<MessageGranule>}} PayloadWithGranules
+*/
+
+/**
+ * @param {unknown} payload
+ * @returns {payload is PayloadWithGranules}
+ */
+function payloadHasGranules(payload) {
+  return (
+    payload instanceof Object
+    && 'granules' in payload
+    && Array.isArray(payload.granules)
+  );
+}
 
+/**
+ * Reformat object with key attributes at top level.
+ *
+ * @param {{[key: string]: any}} messageBody - event bridge event as defined in aws-lambda
+ * @returns {Promise<Object>} - message packaged with
+ * metadata or 'unknown' where metadata not found
+ * {
+ *   error: <errorString | 'unknown'>
+ *   collection: <collectionName | 'unknown'>
+ *   granules: <[granuleIds, ...] | 'unknown'>
+ *   execution: <executionArn | 'unknown'>
+ *   stateMachine: <stateMachineArn | 'unknown'>
+ *   ...originalAttributes
+ * }
+ */
+async function hoistCumulusMessageDetails(messageBody) {
+  const execution = messageBody?.detail?.executionArn || 'unknown';
+  const stateMachine = messageBody?.detail?.stateMachineArn || 'unknown';
+
+  let cumulusMessage;
+  try {
+    if (isEventBridgeEvent(messageBody)) {
+      cumulusMessage = await getCumulusMessageFromExecutionEvent(messageBody);
+    } else {
+      cumulusMessage = null;
+    }
+  } catch {
+    cumulusMessage = null;
+  }
+
+  const collection = cumulusMessage?.meta?.collection?.name || 'unknown';
+  let granules;
+
+  const payload = cumulusMessage?.payload;
+  if (payloadHasGranules(payload)) {
+    granules = payload.granules.map((granule) => granule?.granuleId || 'unknown');
+  } else {
+    granules = 'unknown';
+  }
+
+  return {
+    ...messageBody,
+    collection,
+    granules,
+    execution,
+    stateMachine,
+  };
+}
+/**
+ * Determine execution name from body
+ *
+ * @param {Object} cumulusMessageObject - cumulus message
+ * @returns {string} - <executionName | 'unknown'>
+ */
 function determineExecutionName(cumulusMessageObject) {
```

Comment:
How different is the executionArn from the executionName? If they're the same, I'm wondering why we're grabbing them in different ways. If they're different, I'm wondering why we're grabbing both and if we should consolidate and pick the most useful one. Also thinking we should consolidate by using this method in the above function or eliminating this method and grabbing the info out of the massagedMessage/workedMessage in the handler below. All depends on the answer to the beginning question though.

Reply:
the difference is small:
in my function the executionArn is what I'm grabbing because it seems more useful to debugging,
int he existing code executionName is used to to name the s3 file being created.
do we risk breaking something if we change the s3 file to use arn vs name? (I assume we do)

Reply:
I guess you made the change to check if it would break something or not? Just to be clear, I was not trying to say the change _had to_ be made, I just didn't know the difference and was wondering if it made sense to consolidate. I think the change looks good though as long as nothing breaks.